sdlib/d/gc/arena.d

```diff
@@ -374,6 +428,29 @@ private:
 		unusedExtents.insert(e);
 	}
 
+	void shrinkAllocImpl(Extent* e, uint n, uint oldPages, uint delta) {
```

Comment:
This could get oldPages from the extent. the value is either wrong or redundant. If you are going to pass it down, you got to assert its value is correct at every step of the way.